branch to setup heatmap related scripts

edgargabriel opened the pull request:
## Motivation

Create a branch to collect all the items required for generating heatmap reports

## Technical Details

- add an option to report the bw as unified across all threads in the functional tester.
- add a separate testscript which can be used to run the heatmap related benchmarks. This requires a few additional arguments, and to omit a few (e.g. timeout)

## Submission Checklist

- [ ] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

edgargabriel committed:
add heatmap tester scripts

- add also an option to report the bw as unified across all threads
  in the functional tester.

edgargabriel committed:
add first version of script to generate xlsx file

from the raw data files generated when running the functional tests.

edgargabriel committed:
next iteration of parser script

Yiltan commented on tests/functional_tests/tester.cpp:
```diff
@@ -595,6 +595,10 @@ void Tester::print(uint64_t size) {
   }
 
+  if (args.unified_msg_size) {
+    size *= (args.num_wgs * args.wg_size);
```
Does this impact how we print out bandwidth/latency?

Size is assumed to be per thread here: https://github.com/edgargabriel/rocSHMEM/blob/67591fb3421612a883a0c7f1fbd7988563a396ac/tests/functional_tests/tester.cpp#L605 

Since we multiply it by the number of total threads to get our total size

Yiltan commented on scripts/functional_tests/heatmap-driver.sh:
```diff
@@ -0,0 +1,264 @@
+  PRETTY_PASSED="PASSED"
+fi
+
+# This names/values should match the TestType enum in rocSHMEM/tests/functional_tests/tester.hpp
```
Could we have this in a common file that is shared with the other driver? Or update the comment to indicate the three files that must be kept in sync

edgargabriel replied:
I've updated the comments for now.

abouteiller committed:
Communication volume was already computed correctly, just adjust
printouts for -u

edgargabriel committed:
update benchmark limits

edgargabriel committed:
update comment to indicate that we need to keep

three files in sync

avinashkethineedi commented on tests/functional_tests/tester_arguments.cpp:
```diff
@@ -47,6 +47,9 @@ TesterArguments::TesterArguments(int argc, char *argv[]) {
     } else if (arg == "-a") {
       i++;
       algorithm = atoi(argv[i]);
+    } else if (arg == "-u") {
```
Update  [show_usage](https://github.com/ROCm/rocSHMEM/blob/develop/tests/functional_tests/tester_arguments.cpp#L153) to include this flag.

abouteiller committed:
Revamp the -u option to account for the actual concurrency of calls in
wg/wave primitives
change latency to reflect per-procedure-call cost, as seen from the
caller thread (not reciprocal of message rate)

abouteiller committed:
Change MR test case to go higher message sdizes

Yiltan committed:
update alltoall

Yiltan committed:
HeatMapColl typo

Yiltan committed:
update put/get sizes

abouteiller committed:
Run only 1MB for w1z1 put/get

edgargabriel committed:
update heatmap xlsx generation

abouteiller committed:
Compute volume and max_msg_size correctly for alltoall in -u mode

abouteiller committed:
Add a noverif option to the functional test for cases where verif takes
a long time

abouteiller committed:
Compute volumes for agg bw reporting for all collective tests

Yiltan commented on tests/functional_tests/tester.cpp:
```diff
@@ -123,13 +123,20 @@ Tester::Tester(TesterArguments args) : args(args) {
       case AllToAllsTestType:
       case TeamAlltoallmemOnStreamTestType:
-        max_msg_size = args.max_msg_size / args.num_wgs / args.numprocs;
+        max_msg_size = args.max_msg_size / args.num_wgs / args.numprocs / args.numprocs;
```
```suggestion
      case TeamBroadcastTestType:
      case TeamReductionTestType:
      case TeamFCollectTestType:
      case CollectTestType:
      case TeamAllToAllTestType:
      case AllToAllsTestType:
      case TeamAlltoallmemOnStreamTestType:
        max_msg_size = args.max_msg_size / args.num_wgs / args.numprocs;
```

Yiltan commented on tests/functional_tests/team_reduction_tester.cpp:
```diff
@@ -125,7 +126,7 @@ TeamReductionTester<T1, T2>::~TeamReductionTester() {
 template <typename T1, ROCSHMEM_OP T2>
 void TeamReductionTester<T1, T2>::preLaunchKernel() {
-  int n_pes = rocshmem_team_n_pes(ROCSHMEM_TEAM_WORLD);
+  bw_factor = n_pes;
```
```suggestion
  bw_factor = (int)(2.0 * ((double)n_pes - 1) /  (double)n_pes);
  ```

Yiltan commented on tests/functional_tests/team_broadcast_tester.cpp:
```diff
@@ -138,6 +138,8 @@ TeamBroadcastTester<T1>::~TeamBroadcastTester() {
 
 template <typename T1>
 void TeamBroadcastTester<T1>::preLaunchKernel() {
+  bw_factor = n_pes;
```
```suggestion
  bw_factor = 1;
  ```

Yiltan commented on tests/functional_tests/team_alltoallmem_on_stream_tester.cpp:
```diff
@@ -83,7 +83,7 @@ TeamAlltoallmemOnStreamTester::~TeamAlltoallmemOnStreamTester() {
 
 void TeamAlltoallmemOnStreamTester::preLaunchKernel() {
-  bw_factor = n_pes;
+  bw_factor = n_pes * n_pes;
```
```suggestion
  bw_factor = (int)(2.0 * ((double)n_pes - 1) /  (double)n_pes);
  ```

Yiltan commented on tests/functional_tests/team_alltoall_tester.cpp:
```diff
@@ -136,7 +136,7 @@ TeamAlltoallTester<T1>::~TeamAlltoallTester() {
 template <typename T1>
 void TeamAlltoallTester<T1>::preLaunchKernel() {
-  bw_factor = n_pes;
+  bw_factor = n_pes * n_pes;
```
```suggestion
  bw_factor = (int)(2.0 * ((double)n_pes - 1) /  (double)n_pes);
  ```

Yiltan commented on tests/functional_tests/team_fcollect_tester.cpp:
```diff
@@ -151,7 +151,7 @@ TeamFcollectTester<T1>::~TeamFcollectTester() {
 template <typename T1>
 void TeamFcollectTester<T1>::preLaunchKernel() {
-  bw_factor = n_pes;
+  bw_factor = n_pes * 2;
```
```suggestion
  bw_factor = (int)(2.0 * ((double)n_pes - 1) /  (double)n_pes);
  ```

abouteiller committed:
Update volume for alltoall_on_stream

Co-authored-by: Yiltan <ytemucin@amd.com>

abouteiller committed:
Report rccl like message sizes in all2all

Yiltan committed:
Add missing break

abouteiller committed:
Remove .rej file

abouteiller committed:
Rename -u to -v to control max per-pe volume (-v X equivalent to setting -u 1
-s X)

abouteiller committed:
Reconverge the heatmap-driver and the driver.sh scripts

abouteiller committed:
If we find heatmapcoll do not prefer *coll

ammallya commented:
Imported to ROCm/rocm-systems pr 2735
